modules/security/pam.nix

```
@@ -15,48 +15,86 @@ let
-        ${sed} -i '/${option}/d' ${file}
-      fi
-    ''}
-  '';
+  mkSudoTouchIdAuthScript = cfg:
+    let
+      disableTmuxTouchIdAuth = ''
+        # Disable tmux Touch ID authentication, if added by nix-darwin
+        if grep '${optionTmux}' ${file} > /dev/null; then
+          ${sed} -i '/${optionTmux}/d' ${file}
+        fi
+      '';
+      file = cfg.sudoFile;
+      optionSudo = "security.pam.enableSudoTouchIdAuth";
+      optionTmux = "security.pam.enableTmuxTouchIdAuth";
+      sed = "${pkgs.gnused}/bin/sed";
+    in
+    ''
+      ${if cfg.enableSudoTouchIdAuth then ''
+        # Enable sudo Touch ID authentication, if not already enabled
+        if ! grep 'pam_tid.so' ${file} > /dev/null; then
+          ${sed} -i '2i\
+        auth       sufficient     pam_tid.so # nix-darwin: ${optionSudo}
+          ' ${file}
+        fi
+        ${if cfg.enableTmuxTouchIdSupport then ''
+        # Enable tmux Touch ID authentication, if not already enabled
+        if ! grep 'pam_reattach.so' ${file} > /dev/null; then
+          ${sed} -i '2i\
+        auth       optional       ${pkgs.pam-reattach}/lib/pam/pam_reattach.so # nix-darwin: ${optionTmux}
+          ' ${file}
+        fi
+        '' else disableTmuxTouchIdAuth
+        }
+      '' else ''
+        ${disableTmuxTouchIdAuth}
+        # Disable sudo Touch ID authentication, if added by nix-darwin
+        if grep '${optionSudo}' ${file} > /dev/null; then
+          ${sed} -i '/${optionSudo}/d' ${file}
+        fi
+      ''}
+    '';
 in
 
 {
   options = {
-    security.pam.enableSudoTouchIdAuth = mkEnableOption ''
-      Enable sudo authentication with Touch ID
+    security.pam = {
+      enableSudoTouchIdAuth = mkEnableOption ''
+        Enable sudo authentication with Touch ID
 
-      When enabled, this option adds the following line to /etc/pam.d/sudo:
+        When enabled, this option adds the following line to /etc/pam.d/sudo:
 
-          auth       sufficient     pam_tid.so
+            auth       sufficient     pam_tid.so
 
-      (Note that macOS resets this file when doing a system update. As such, sudo
-      authentication with Touch ID won't work after a system update until the nix-darwin
-      configuration is reapplied.)
-    '';
+        (Note that macOS resets this file when doing a system update. As such, sudo
+        authentication with Touch ID won't work after a system update until the nix-darwin
+        configuration is reapplied.)
+      '';
+      enableTmuxTouchIdSupport = mkEnableOption ''
+        Enable sudo authentication with Touch ID inside tmux
+
+        When enabled, this option adds the following line to /etc/pam.d/sudo:
+
+            auth       optional     pam_reattach.so
+
+        (Note that macOS resets this file when doing a system update. As such,
+        sudo authentication with Touch ID inside tmux won't work after a system
+        update until the nix-darwin configuration is reapplied.)
```

While this is primarily useful for tmux (and therefore I agree that tmux support should be mentioned in the description/documentation), I don't think the option itself should have "tmux" in the name:

```suggestion
      enablePamReattach = mkEnableOption ''
        Enable sudo authentication with Touch ID within programs running in the background session by reattaching to the authenticating user's per-session bootstrap namespace.
        
        This allows users to make use of the pam_tid module (Touch ID) from within tmux.

        When enabled, this option adds the following line to /etc/pam.d/sudo:

            auth       optional     pam_reattach.so

        (Note that macOS resets this file when doing a system update. As such,
        sudo authentication with Touch ID inside tmux won't work after a system
        update until the nix-darwin configuration is reapplied.)
```